### PR TITLE
Log file names and line numbers

### PR DESCRIPTION
**Pull request by TheoBerlin**
## Purpose
Instead of manually typing the file name when logging, as in:
`LOG_ERROR("[BufferVK]: Failed to allocate memory");`

File names and line numbers are now automatically printed, so
`LOG_ERROR("Failed to allocate memory");`
prints
`BufferVK.cpp (105): Created Buffer`

This PR also replaces the various debug logs `D_LOG_`... with `LOG_DEBUG`. Also, PhysX error callbacks now include file names and line numbers.

Also removed a line that adds a parent component to the player camera entity. This caused an assertion to trigger in debug builds.

## Changes
The various `LOG` macros now utilize the compile-time variables `__FILE__` and `__LINE__`.

## Testing
How have one tested the feature to ensure it works?
- [x] Tested in Sandbox
- [ ] Tested in Multiplayer with 2 clients
- [ ] Tested in Multiplayer with more than 2 clients
- [ ] Tested in Benchmark

### CODE REVIEW

File: CrazyCanvas/Source/States/SandboxState.cpp [hunk 1]
```diff
@@ -358,7 +358,7 @@ void SandboxState::Init()
 	//	pECS->AddComponent<RotationComponent>(m_DirLight, { true, glm::quatLookAt(glm::normalize(glm::vec3(0.5f, -1.0f, 0.5f)), g_DefaultUp) });
 	//	pECS->AddComponent<DirectionalLightComponent>(m_DirLight, directionalLightComponent);
 
-	//	D_LOG_INFO("[LevelObjectCreator]: Created Directional Light");
+	//	LOG_DEBUG("[LevelObjectCreator]: Created Directional Light");
```
Comment (TheRealChriAnd) [inline CrazyCanvas/Source/States/SandboxState.cpp]:
```suggestion
	//	LOG_DEBUG("Created Directional Light");
```

File: CrazyCanvas/Source/World/LevelObjectCreator.cpp [hunk 2]
```diff
@@ -156,7 +156,7 @@ LambdaEngine::Entity LevelObjectCreator::CreateDirectionalLight(
 		pECS->AddComponent<RotationComponent>(entity, { true, glm::quatLookAt({directionalLight.Direction}, g_DefaultUp) });
 		pECS->AddComponent<DirectionalLightComponent>(entity, directionalLightComponent);
 
-		D_LOG_INFO("[LevelObjectCreator]: Created Directional Light");
+		LOG_DEBUG("[LevelObjectCreator]: Created Directional Light");
```
Comment (TheRealChriAnd) [inline CrazyCanvas/Source/World/LevelObjectCreator.cpp]:
```suggestion
		LOG_DEBUG("[LevelObjectCreator]: Created Directional Light");
```
```suggestion
		LOG_DEBUG("Created Directional Light");
```

File: CrazyCanvas/Source/World/LevelObjectCreator.cpp [hunk 3]
```diff
@@ -181,7 +181,7 @@ LambdaEngine::Entity LevelObjectCreator::CreatePointLight(const LambdaEngine::Lo
 		pECS->AddComponent<PositionComponent>(entity, { true, (pointLight.Position + translation) });
 		pECS->AddComponent<PointLightComponent>(entity, pointLightComponent);
 
-		D_LOG_INFO("[LevelObjectCreator]: Created Point Light");
+		LOG_DEBUG("[LevelObjectCreator]: Created Point Light");
```
Comment (TheRealChriAnd) [inline CrazyCanvas/Source/World/LevelObjectCreator.cpp]:
```suggestion
		LOG_DEBUG("Created Point Light");
```

File: LambdaEngine/Source/Application/Win32/Win32Application.cpp [hunk 4]
```diff
@@ -644,7 +644,7 @@ namespace LambdaEngine
 		}
 		else
 		{
-			D_LOG_MESSAGE("[Win32Application]: Registered Raw Input devices");
+			LOG_DEBUG("[Win32Application]: Registered Raw Input devices");
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Application/Win32/Win32Application.cpp]:
```suggestion
			LOG_DEBUG("Registered Raw Input devices");
```

File: LambdaEngine/Source/Application/Win32/Win32Application.cpp [hunk 5]
```diff
@@ -669,7 +669,7 @@ namespace LambdaEngine
 		}
 		else
 		{
-			D_LOG_MESSAGE("[Win32Application]: Unregistered Raw Input devices");
+			LOG_DEBUG("[Win32Application]: Unregistered Raw Input devices");
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Application/Win32/Win32Application.cpp]:
```suggestion
			LOG_DEBUG("Unregistered Raw Input devices");
```

File: LambdaEngine/Source/Audio/FMOD/AudioDeviceFMOD.cpp [hunk 6]
```diff
@@ -25,28 +25,28 @@ namespace LambdaEngine
 		}
 		else if (!(severity & FMOD_DEBUG_LEVEL_LOG))
 		{
-			D_LOG_INFO("[FMOD_DEBUG]: [%s : %u : %s] - \"%s\"", pFile, line, pFunc, pMessage);
+			LOG_DEBUG("[FMOD_DEBUG]: [%s : %u : %s] - \"%s\"", pFile, line, pFunc, pMessage);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Audio/FMOD/AudioDeviceFMOD.cpp]:
```suggestion
			LOG_DEBUG("[%s : %u : %s] - \"%s\"", pFile, line, pFunc, pMessage);
```

File: LambdaEngine/Source/Audio/FMOD/AudioGeometryFMOD.cpp [hunk 7]
```diff
@@ -79,7 +79,7 @@ namespace LambdaEngine
 			}
 		}
 
-		D_LOG_MESSAGE("[AudioGeometryFMOD]: Successfully initialized %s!", m_pName);
+		LOG_DEBUG("[AudioGeometryFMOD]: Successfully initialized %s!", m_pName);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Audio/FMOD/AudioGeometryFMOD.cpp]:
```suggestion
		LOG_DEBUG("Successfully initialized %s!", m_pName);
```

File: LambdaEngine/Source/Audio/FMOD/MusicFMOD.cpp [hunk 8]
```diff
@@ -67,7 +67,7 @@ namespace LambdaEngine
 
 		if (FMOD_Channel_SetVolume(m_pChannel, volume) != FMOD_OK)
 		{
-			D_LOG_WARNING("[MusicFMOD]: Volume could not be set for %s", m_Name.c_str());
+			LOG_DEBUG("[MusicFMOD]: Volume could not be set for %s", m_Name.c_str());
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Audio/FMOD/MusicFMOD.cpp]:
```suggestion
			LOG_DEBUG("Volume could not be set for %s", m_Name.c_str());
```

File: LambdaEngine/Source/Audio/FMOD/MusicFMOD.cpp [hunk 9]
```diff
@@ -77,7 +77,7 @@ namespace LambdaEngine
 
 		if (FMOD_Channel_SetPitch(m_pChannel, pitch) != FMOD_OK)
 		{
-			D_LOG_WARNING("[MusicFMOD]: Pitch could not be set for %s", m_Name.c_str());
+			LOG_DEBUG("[MusicFMOD]: Pitch could not be set for %s", m_Name.c_str());
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Audio/FMOD/MusicFMOD.cpp]:
```suggestion
			LOG_DEBUG("Pitch could not be set for %s", m_Name.c_str());
```

File: LambdaEngine/Source/Audio/FMOD/ReverbSphereFMOD.cpp [hunk 10]
```diff
@@ -40,7 +40,7 @@ namespace LambdaEngine
 
 		Set3DAttributes(pDesc->Position, pDesc->MinDistance, pDesc->MaxDistance);
 
-		D_LOG_MESSAGE("[ReverbSphereFMOD]: Successfully initialized %s!", m_pName);
+		LOG_DEBUG("[ReverbSphereFMOD]: Successfully initialized %s!", m_pName);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Audio/FMOD/ReverbSphereFMOD.cpp]:
```suggestion
		LOG_DEBUG("Successfully initialized %s!", m_pName);
```

File: LambdaEngine/Source/Audio/FMOD/SoundInstance2DFMOD.cpp [hunk 11]
```diff
@@ -86,7 +86,7 @@ namespace LambdaEngine
 		{
 			if (FMOD_Channel_SetVolume(m_pChannel, volume) != FMOD_OK)
 			{
-				D_LOG_WARNING("[SoundInstance2DFMOD]: Volume could not be set for %s", m_pName);
+				LOG_DEBUG("[SoundInstance2DFMOD]: Volume could not be set for %s", m_pName);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Audio/FMOD/SoundInstance2DFMOD.cpp]:
```suggestion
				LOG_DEBUG("Volume could not be set for %s", m_pName);
```

File: LambdaEngine/Source/Audio/FMOD/SoundInstance2DFMOD.cpp [hunk 12]
```diff
@@ -99,7 +99,7 @@ namespace LambdaEngine
 		{
 			if (FMOD_Channel_SetPitch(m_pChannel, pitch) != FMOD_OK)
 			{
-				D_LOG_WARNING("[SoundInstance2DFMOD]: Pitch could not be set for %s", m_pName);
+				LOG_DEBUG("[SoundInstance2DFMOD]: Pitch could not be set for %s", m_pName);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Audio/FMOD/SoundInstance2DFMOD.cpp]:
```suggestion
				LOG_DEBUG("Pitch could not be set for %s", m_pName);
```

File: LambdaEngine/Source/Audio/FMOD/SoundInstance3DFMOD.cpp [hunk 13]
```diff
@@ -90,7 +90,7 @@ namespace LambdaEngine
 
 			if (FMOD_Channel_Set3DAttributes(m_pChannel, &fmodPosition, nullptr) != FMOD_OK)
 			{
-				D_LOG_WARNING("[SoundInstance3DFMOD]: 3D Attributes could not be set for %s", m_pName);
+				LOG_DEBUG("[SoundInstance3DFMOD]: 3D Attributes could not be set for %s", m_pName);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Audio/FMOD/SoundInstance3DFMOD.cpp]:
```suggestion
				LOG_DEBUG("[SoundInstance3DFMOD]: 3D Attributes could not be set for %s", m_pName);
```
```suggestion
				LOG_DEBUG("3D Attributes could not be set for %s", m_pName);
```

File: LambdaEngine/Source/Audio/FMOD/SoundInstance3DFMOD.cpp [hunk 14]
```diff
@@ -103,7 +103,7 @@ namespace LambdaEngine
 		{
 			if (FMOD_Channel_SetVolume(m_pChannel, volume) != FMOD_OK)
 			{
-				D_LOG_WARNING("[SoundInstance3DFMOD]: Volume could not be set for %s", m_pName);
+				LOG_DEBUG("[SoundInstance3DFMOD]: Volume could not be set for %s", m_pName);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Audio/FMOD/SoundInstance3DFMOD.cpp]:
```suggestion
				LOG_DEBUG("Volume could not be set for %s", m_pName);
```

File: LambdaEngine/Source/Audio/FMOD/SoundInstance3DFMOD.cpp [hunk 15]
```diff
@@ -116,7 +116,7 @@ namespace LambdaEngine
 		{
 			if (FMOD_Channel_SetPitch(m_pChannel, pitch) != FMOD_OK)
 			{
-				D_LOG_WARNING("[SoundInstance3DFMOD]: Pitch could not be set for %s", m_pName);
+				LOG_DEBUG("[SoundInstance3DFMOD]: Pitch could not be set for %s", m_pName);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Audio/FMOD/SoundInstance3DFMOD.cpp]:
```suggestion
				LOG_DEBUG("Pitch could not be set for %s", m_pName);
```

File: LambdaEngine/Source/Debug/GPUProfiler.cpp [hunk 16]
```diff
@@ -325,7 +325,7 @@ namespace LambdaEngine
 		}
 		else
 		{
-			D_LOG_ERROR("[GPUProfiler]: Failed to get available results with %d number of timestamps", timestampCount);
+			LOG_DEBUG("[GPUProfiler]: Failed to get available results with %d number of timestamps", timestampCount);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Debug/GPUProfiler.cpp]:
```suggestion
			LOG_DEBUG("Failed to get available results with %d number of timestamps", timestampCount);
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/CommandQueueVK.cpp [hunk 17]
```diff
@@ -44,8 +44,8 @@ namespace LambdaEngine
 		{
 			vkGetDeviceQueue(m_pDevice->Device, queueFamilyIndex, index, &m_Queue);
 			SetName(debugName);
-			
-			D_LOG_MESSAGE("[CommandQueueVK]: Created commandqueue from queuefamily=%u with index=%u", queueFamilyIndex, index);
+
+			LOG_DEBUG("[CommandQueueVK]: Created commandqueue from queuefamily=%u with index=%u", queueFamilyIndex, index);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/CommandQueueVK.cpp]:
```suggestion
			LOG_DEBUG("Created commandqueue from queuefamily=%u with index=%u", queueFamilyIndex, index);
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/DeviceAllocatorVK.cpp [hunk 18]
```diff
@@ -385,7 +385,7 @@ namespace LambdaEngine
 				{
 					if ((pIterator->Offset + pIterator->TotalSizeInBytes) > (pIterator->pNext->Offset))
 					{
-						D_LOG_WARNING("[DeviceMemoryPageVK]: Overlap found");
+						LOG_DEBUG("[DeviceMemoryPageVK]: Overlap found");
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/DeviceAllocatorVK.cpp]:
```suggestion
						LOG_DEBUG("Overlap found");
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/FenceVK.cpp [hunk 19]
```diff
@@ -83,31 +83,31 @@ namespace LambdaEngine
 				m_Fences[i] = fence;
 			}
 		}
-		
-		D_LOG_MESSAGE("[FenceVK]: Created Fence");
-		
+
+		LOG_DEBUG("[FenceVK]: Created Fence");
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/FenceVK.cpp]:
```suggestion
		LOG_DEBUG("Created Fence");
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/FenceVK.cpp [hunk 20]
```diff
@@ -292,7 +292,7 @@ namespace LambdaEngine
 			}
 			else
 			{
-				D_LOG_WARNING("[FenceVK]: New fence created");
+				LOG_DEBUG("[FenceVK]: New fence created");
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/FenceVK.cpp]:
```suggestion
				LOG_DEBUG("New fence created");
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/FenceVK.cpp [hunk 21]
```diff
@@ -322,7 +322,7 @@ namespace LambdaEngine
 			}
 			else
 			{
-				D_LOG_WARNING("[FenceVK]: New semaphore created");
+				LOG_DEBUG("[FenceVK]: New semaphore created");
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/FenceVK.cpp]:
```suggestion
				LOG_DEBUG("New semaphore created");
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/FrameBufferCacheVK.cpp [hunk 22]
```diff
@@ -112,7 +112,7 @@ namespace LambdaEngine
 		}
 		else
 		{
-			//D_LOG_MESSAGE("[FrameBufferCacheVK]: Created framebuffer [0]:%p [1]:%p [2]:%p [3]:%p [4]:%p [5]:%p [6]:%p [7]:%p [Depth]:%p [Pass]:%p", key.ColorAttachmentsViews[0], key.ColorAttachmentsViews[1], key.ColorAttachmentsViews[2], key.ColorAttachmentsViews[3], key.ColorAttachmentsViews[4], key.ColorAttachmentsViews[5], key.ColorAttachmentsViews[6], key.ColorAttachmentsViews[7], key.DepthStencilView, key.RenderPass);
+			//LOG_DEBUG("[FrameBufferCacheVK]: Created framebuffer [0]:%p [1]:%p [2]:%p [3]:%p [4]:%p [5]:%p [6]:%p [7]:%p [Depth]:%p [Pass]:%p", key.ColorAttachmentsViews[0], key.ColorAttachmentsViews[1], key.ColorAttachmentsViews[2], key.ColorAttachmentsViews[3], key.ColorAttachmentsViews[4], key.ColorAttachmentsViews[5], key.ColorAttachmentsViews[6], key.ColorAttachmentsViews[7], key.DepthStencilView, key.RenderPass);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/FrameBufferCacheVK.cpp]:
```suggestion
			//LOG_DEBUG("Created framebuffer [0]:%p [1]:%p [2]:%p [3]:%p [4]:%p [5]:%p [6]:%p [7]:%p [Depth]:%p [Pass]:%p", key.ColorAttachmentsViews[0], key.ColorAttachmentsViews[1], key.ColorAttachmentsViews[2], key.ColorAttachmentsViews[3], key.ColorAttachmentsViews[4], key.ColorAttachmentsViews[5], key.ColorAttachmentsViews[6], key.ColorAttachmentsViews[7], key.DepthStencilView, key.RenderPass);
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/RayTracingPipelineStateVK.cpp [hunk 23]
```diff
@@ -157,11 +157,11 @@ namespace LambdaEngine
 		SetName(pDesc->DebugName);
 		if (!pDesc->DebugName.empty())
 		{
-			D_LOG_MESSAGE("[RayTracingPipelineStateVK]: Created Pipeline for %s", pDesc->DebugName.c_str());
+			LOG_DEBUG("[RayTracingPipelineStateVK]: Created Pipeline for %s", pDesc->DebugName.c_str());
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/RayTracingPipelineStateVK.cpp]:
```suggestion
			LOG_DEBUG("Created Pipeline for %s", pDesc->DebugName.c_str());
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/RayTracingPipelineStateVK.cpp [hunk 24]
```diff
@@ -157,11 +157,11 @@ namespace LambdaEngine
...
 		}
 		else
 		{
-			D_LOG_MESSAGE("[RayTracingPipelineStateVK]: Created Pipeline");
+			LOG_DEBUG("[RayTracingPipelineStateVK]: Created Pipeline");
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/RayTracingPipelineStateVK.cpp]:
```suggestion
			LOG_DEBUG("Created Pipeline");
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/SBTVK.cpp [hunk 25]
```diff
@@ -46,11 +46,11 @@ namespace LambdaEngine
 
 		if (!pDesc->DebugName.empty())
 		{
-			D_LOG_MESSAGE("[SBTVK]: Created SBT %s", pDesc->DebugName.c_str());
+			LOG_DEBUG("[SBTVK]: Created SBT %s", pDesc->DebugName.c_str());
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/SBTVK.cpp]:
```suggestion
			LOG_DEBUG("Created SBT %s", pDesc->DebugName.c_str());
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/SBTVK.cpp [hunk 26]
```diff
@@ -46,11 +46,11 @@ namespace LambdaEngine
...
 		}
 		else
 		{
-			D_LOG_MESSAGE("[SBTVK]: Created SBT");
+			LOG_DEBUG("[SBTVK]: Created SBT");
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/SBTVK.cpp]:
```suggestion
			LOG_DEBUG("Created SBT");
```

File: LambdaEngine/Source/Rendering/Core/Vulkan/TextureVK.cpp [hunk 27]
```diff
@@ -137,17 +137,17 @@ namespace LambdaEngine
...
 
 		m_Image = image;
 		m_Desc = *pDesc;
-		
+
 		SetName(m_Desc.DebugName);
-		
-		D_LOG_MESSAGE("[TextureVK]: Created texture w=%d, h=%d, d=%d", pDesc->Width, pDesc->Height, pDesc->Depth);
+
+		LOG_DEBUG("[TextureVK]: Created texture w=%d, h=%d, d=%d", pDesc->Width, pDesc->Height, pDesc->Depth);
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/Core/Vulkan/TextureVK.cpp]:
```suggestion
		LOG_DEBUG("Created texture w=%d, h=%d, d=%d", pDesc->Width, pDesc->Height, pDesc->Depth);
```

File: LambdaEngine/Source/Rendering/ParticleCollider.cpp [hunk 28]
```diff
@@ -188,7 +188,7 @@ namespace LambdaEngine
 		m_InlineRayTracingEnabled = RenderSystem::GetInstance().IsInlineRayTracingEnabled();
 		if (!m_InlineRayTracingEnabled)
 		{
-			D_LOG_INFO("[ParticleCollider] Ray query is disabled, disable particle collisions.");
+			LOG_DEBUG("[ParticleCollider] Ray query is disabled, disable particle collisions.");
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/ParticleCollider.cpp]:
```suggestion
			LOG_DEBUG("Ray query is disabled, disable particle collisions.");
```

File: LambdaEngine/Source/Rendering/PipelineStateManager.cpp [hunk 29]
```diff
@@ -143,7 +143,7 @@ namespace LambdaEngine
 		}
 		else
 		{
-			D_LOG_ERROR("[PipelineStateManager]: PipelineState is nullptr");
+			LOG_DEBUG("[PipelineStateManager]: PipelineState is nullptr");
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/PipelineStateManager.cpp]:
```suggestion
			LOG_DEBUG("PipelineState is nullptr");
```

File: LambdaEngine/Source/Rendering/PipelineStateManager.cpp [hunk 30]
```diff
@@ -161,7 +161,7 @@ namespace LambdaEngine
...
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/PipelineStateManager.cpp]:
```suggestion
			LOG_DEBUG("PipelineState is nullptr");
```

File: LambdaEngine/Source/Rendering/PipelineStateManager.cpp [hunk 31]
```diff
@@ -179,7 +179,7 @@ namespace LambdaEngine
...
```
Comment (TheRealChriAnd) [inline LambdaEngine/Source/Rendering/PipelineStateManager.cpp]:
```suggestion
			LOG_DEBUG("PipelineState is nullptr");
```

**Review by TheRealChriAnd — CHANGES_REQUESTED** [review]
Looking great, but please fix the duplicated prints

**Comment by TheoBerlin** [comment]
> Looking great, but please fix the duplicated prints

Done

**Review by TheRealChriAnd — APPROVED** [review]

**Pull request merged**